mydumper.c

```diff
@@ -1796,7 +1796,7 @@ MYSQL_STMT *execute_detect_fields_stmt(MYSQL *conn, char *database, char *table,
 }
 
 gboolean detect_generated_fields(MYSQL *conn, char *database, char *table){
-	const char* query = "select 1 from information_schema.COLUMNS where TABLE_SCHEMA=? and TABLE_NAME=? and extra like '%GENERATED%'";
+	const char* query = "select 1 from information_schema.COLUMNS where TABLE_SCHEMA=? and TABLE_NAME=? and extra REGEXP 'STORED|VIRTUAL GENERATED'";
```

Comment:
Should it be `VIRTUAL (GENERATED|STORED)`? So it won't just match a plain `STORED` without the `VIRTUAL`.

Reply:
yep, `(STORED|VIRTUAL) GENERATED` is more accurate.